net/packet/checksum/checksum_test.go

```diff
@@ -124,25 +125,43 @@ func TestNatChecksumsV6UDP(t *testing.T) {
 	}
 
 	// Parse the packet.
-	var p packet.Parsed
+	var p, p2 packet.Parsed
 	p.Decode(b)
 	t.Log(p.String())
 
 	// Update the source address of the packet to be the same as the dest.
 	UpdateSrcAddr(&p, a2)
+	p2.Decode(p.Buffer())
+	if p2.Src.Addr() != a2 {
+		t.Fatalf("got %v, want %v", p2.Src, a2)
+	}
 	if !udp.IsChecksumValid(tcpip.AddrFrom16Slice(a2.AsSlice()), tcpip.AddrFrom16Slice(a2.AsSlice()), checksum.Checksum(b.Payload()[header.UDPMinimumSize:], 0)) {
 		t.Fatal("incorrect checksum after updating source address")
 	}
 
 	// Update the dest address of the packet to be the original source address.
 	UpdateDstAddr(&p, a1)
+	p2.Decode(p.Buffer())
+	if p2.Dst.Addr() != a1 {
+		t.Fatalf("got %v, want %v", p2.Dst, a1)
+	}
 	if !udp.IsChecksumValid(tcpip.AddrFrom16Slice(a2.AsSlice()), tcpip.AddrFrom16Slice(a1.AsSlice()), checksum.Checksum(b.Payload()[header.UDPMinimumSize:], 0)) {
 		t.Fatal("incorrect checksum after updating destination address")
 	}
 }
 
+func randV6Addr() netip.Addr {
+	a1, a2 := rand.Int64(), rand.Int64()
+	return netip.AddrFrom16([16]byte{
+		byte(a1 >> 56), byte(a1 >> 48), byte(a1 >> 40), byte(a1 >> 32),
+		byte(a1 >> 24), byte(a1 >> 16), byte(a1 >> 8), byte(a1),
+		byte(a2 >> 56), byte(a2 >> 48), byte(a2 >> 40), byte(a2 >> 32),
+		byte(a2 >> 24), byte(a2 >> 16), byte(a2 >> 8), byte(a2),
+	})
+}
```

Comment:
I worry that failures will end up looking like flaky tests rather than consistent failures when using randomized inputs. Also, ideally it should be possible to re-run with the same random seed like fuzzing tests do.

The best option would probably be a random seed that is logged when the test begins and can be overridden with a command line flag. Failing that, using a fixed random seed but running a few times with several generated addresses is probably good.

This will do okay for now though.

Also, I guess it might be good to avoid randomly generating unallocated/reserved addresses? They might have different semantics than the current unicast ranges.

Reply:
it currently prints out the addresses that were used on failure so rerunning with the appropriate input is a matter of copying the addresses out, i agree that a single seed would make it easier but this seems easy enough. If you feel strongly, a compromise could be to add `--src` and `--dst` flags but i'm happy with this as is.

As for unallocated/reserved ranges, this package/func is too low-level to care about those. the API is "change the dst on this packet", not sure if it's worth adding restrictions regarding that here.

Reply:
Nah that's fine for now. Better tooling for random numbers in tests would be nice but not urgent. This is merely one of many situations where that would help.